packages/dids/src/methods/did-dht.ts

```diff
@@ -1260,10 +1260,14 @@ export class DidDhtDocument {
       data : rootRecord.join(PROPERTY_SEPARATOR)
     });
 
-    // Per the DID DHT specification, the method-specific identifier must be appended as the
-    // Origin of all records.
+    // Per the DID DHT specification (https://did-dht.com/#dids-as-dns-records):
+    // Record name of a Root Record or a Authoritative Gateway Records MUST end in `<ID>.`
     const [, , identifier] = didDocument.id.split(':');
-    dnsAnswerRecords.forEach(record => record.name += identifier);
+    dnsAnswerRecords.forEach(record => {
+      if (record.name.startsWith('_did.')) {
+        record.name += identifier + '.';
+      }
+    });
```

Comment:
Instead of appending identifier after creating the `answer`, what do you think of setting it when we create the root record and auth gateway records? For the root record, that would mean updating this line above `name : '_did.'` to be this `name : `_did.${didDocument.id.split(':')[2]}`. i'm not sure where we create auth gateway records, but it would be a similar pattern there.

Reply:
Sounds cleaner! Imma try it!